jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactoryInternals.java

```diff
@@ -93,6 +93,44 @@ protected void onMounted(FileSystemPool.Mount mount, URI uri)
         }
     };
 
+    /**
+     * Get the {@link ResourceFactory} that is registered for the specific scheme.
+     *
+     * <pre>{@code
+     * .byScheme("jar") == ResourceFactory supporting jar
+     * .byScheme("jar:file://foo.jar!/") == null // full url strings not supported)
+     * }</pre>
+     *
+     * @param scheme the scheme to look up
+     * @return the {@link ResourceFactory} responsible for the scheme, null if no {@link ResourceFactory} handles the scheme.
+     * @see ResourceFactory#registerResourceFactory(String, ResourceFactory)
+     * @see ResourceFactory#unregisterResourceFactory(String)
+     * @see #getBestByScheme(String)
+     */
+    static ResourceFactory byScheme(String scheme)
```

Comment:
This is a new concept?   We now have two orthogonal reasons for selecting a resource factory.  Should we select one by lifecycle or by scheme?   Why is this necessary?

Edit: Oh I see it is not a new concept... and at least this is moving it internal.   But I'm struggling to see why we need it?

Reply:
Oh Oh I see we are not actually picking a ResourceFactory by scheme, rather we are checking if we know of the scheme.  Can we change this to a boolean return?